packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js

```diff
@@ -143,6 +166,34 @@ describe('ContactPreferences Component', () => {
             // Assert
             expect(wrapper.vm.isFormDirty).toEqual(false);
         });
+
+        it('should not call initialise method if the authorisation has not completed', () => {
+            // Arrange
+            sutProps = { ...sutProps, isAuthFinished: false };
+
+            // Act
+            wrapper = mountContactPreferences();
+
+            // Assert
+            expect(storeActions.loadPreferences).not.toHaveBeenCalled();
+        });
+
+        it('should only call initialise method once the authorisation has completed', async () => {
```

Comment:
If I understand correctly, I think in this test what you're trying to check is that when your watcher sees that the authorisation is complete, the initialise method is called.

To me, this test would look something like this (kinda pseudo-code):

```
// Arrange
wrapper = mountContactPreferences();

// Act
wrapper.vm.$options.watch.isAuthFinished.call(wrapper.vm, true);

// Assert
expect(initialiseSpy).toHaveBeenCalled();
```
More info: https://github.com/vuejs/vue-test-utils/issues/331#issuecomment-382037200

Reply:
@SachaZvetelman I want to mount the component with the `isAuthFinished` = false then check that the `initialise()` has not been called then set the prop and recheck that the `initialise()` has now been called.

Reply:
If you comment out the watcher, does this test still pass? I'm not sure. If it fails when you do that, I'm ok to approve. Could you please check that?

Reply:
@SachaZvetelman I comment out the watcher and it failed